### PR TITLE
Made the xrootd cache prefetch variable configurable via a pelican va…

### PR DESCRIPTION
**Pull request by turetske**
…riable.

Added the Pelican variable Cache.Prefetch and used it to set the cache prefix value in the cache Changed its default from 20 to 0
Updated the xrootd config and cache templates accordingly

### CODE REVIEW

**Comment by jhiemstrawisc** [comment]
> …riable.
> 
> Added the Pelican variabled Cache.Prefix and used it to set the cache prefix value in the cache Changed its default from 20 to 0 Updated the xrootd config and cache templates accordingly

Just clarifying before I review, is this supposed to be `Cache.Prefix` like the description claims or `Cache.Prefetch` like the title claims?

**Comment by turetske** [comment]
> > …riable.
> > Added the Pelican variabled Cache.Prefix and used it to set the cache prefix value in the cache Changed its default from 20 to 0 Updated the xrootd config and cache templates accordingly
> 
> Just clarifying before I review, is this supposed to be `Cache.Prefix` like the description claims or `Cache.Prefetch` like the title claims?

Prefetch, I had just been dealing with a lot of Prefix stuff and the typo just came out.

I recommend waiting until I can fix the test failures, though.

**Review by jhiemstrawisc — CHANGES_REQUESTED** [review]

File: xrootd/xrootd_config.go [hunk 1]
```diff
@@ -117,6 +117,7 @@ type (
 		MetaLocations  []string
 		LocalRoot      string
 		PSSOrigin      string
+		Prefetch       int
```
[the remark this answers is not in the document]
Comment (turetske) [reply xrootd/xrootd_config.go:120]: This arose from when the S3 origins were being pummeled by the cache. One thing we realized is that we didn't want to prefetch blocks when just doing a stat request.

**Review by jhiemstrawisc — APPROVED** [review]
LGTM. I'll approve pending the failing ubuntu tests (not sure if this is a known flaky test or not).

**Pull request merged**